internal/service/encryptionatrest/resource_encryption_at_rest.go

```diff
@@ -214,21 +214,16 @@ func (r *encryptionAtRestRS) Create(ctx context.Context, req resource.CreateRequ
 	}
 
 	projectID := encryptionAtRestPlan.ProjectID.ValueString()
-	encryptionAtRestReq := &admin.EncryptionAtRest{}
-	if encryptionAtRestPlan.AwsKmsConfig != nil {
-		encryptionAtRestReq.AwsKms = newAtlasAwsKms(encryptionAtRestPlan.AwsKmsConfig)
-	}
-	if encryptionAtRestPlan.AzureKeyVaultConfig != nil {
-		encryptionAtRestReq.AzureKeyVault = newAtlasAzureKeyVault(encryptionAtRestPlan.AzureKeyVaultConfig)
-	}
-	if encryptionAtRestPlan.GoogleCloudKmsConfig != nil {
-		encryptionAtRestReq.GoogleCloudKms = newAtlasGcpKms(encryptionAtRestPlan.GoogleCloudKmsConfig)
+	encryptionAtRestReq := &admin.EncryptionAtRest{
+		AwsKms:         NewAtlasAwsKms(encryptionAtRestPlan.AwsKmsConfig),
+		AzureKeyVault:  NewAtlasAzureKeyVault(encryptionAtRestPlan.AzureKeyVaultConfig),
+		GoogleCloudKms: NewAtlasGcpKms(encryptionAtRestPlan.GoogleCloudKmsConfig),
```

Comment:
is there a risk in this change? As could potentially send properties with empty object.

Reply:
In the 3 methods we are already doing this check 
```
if len(tfAzKeyVaultConfigSlice) == 0 {
	return &admin.AzureKeyVault{}
}
```
If it's nil, length is 0 and we set the empty object, which would be equivalent to not setting the value, right?

Reply:
@AgustinBettati I double checked and you are right, we would be setting it with empty object instead of nil. I changed it back to check for nil before calling the methods to set the attributes. Thanks